File: java/입문/옷가게_할인_받기.java
class Solution {
    public int solution(int price) {
        int answer = price;
        final int DCA = 500_000;
        final int DCB = 300_000;
        final int DCC = 100_000;
        
        if (price >= DCA) {
            answer = (int) (price * 0.8);
        } else if (price >= DCB) {
            answer = (int) (price * 0.9);
        } else if (price >= DCC) {
            answer = (int) (price * 0.95);
        }
    
        return answer;
    }
}

File: swift/입문/옷가게_할인_받기.swift
import Foundation

func solution(_ price:Int) -> Int {
    guard price >= 10, price <= 1_000_000 else {
        return 0
    }
    if price >= 500_000 {
        return Int(Double(price) * 0.8)
    } else if price >= 300_000 {
        return Int(Double(price) * 0.9)
    } else if price >= 100_000 {
        return Int(Double(price) * 0.95)
    }
    return price
}